gateway/http/middleware/logger.go

```diff
@@ -0,0 +1,94 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/julienstroheker/AzHexGate/internal/logging"
+)
+
+// responseWriter is a wrapper around http.ResponseWriter that captures the status code
+type responseWriter struct {
+	http.ResponseWriter
+
+	statusCode int
+	written    bool
+}
+
+func (rw *responseWriter) WriteHeader(code int) {
+	if !rw.written {
+		rw.statusCode = code
+		rw.written = true
+		rw.ResponseWriter.WriteHeader(code)
+	}
+}
+
+func (rw *responseWriter) Write(b []byte) (int, error) {
+	if !rw.written {
+		rw.WriteHeader(http.StatusOK)
+	}
+	return rw.ResponseWriter.Write(b)
+}
+
+// Logger is a middleware that logs HTTP requests and responses
+// It logs when a request is received and when the response is sent
+func Logger(logger *logging.Logger) func(http.Handler) http.Handler {
```

Comment:
Is this not an issue passing the logger here ? What if we add fields and change the logger down in the handlers or anywhere else? Is this gonna be reflected here? Is that not better to make sure the logger is been propagated in the context and retrieved from it when needed ?